app/app.go

```diff
@@ -377,7 +377,7 @@ func New(
 	ibcRouter.AddRoute(icacontrollertypes.SubModuleName, icaControllerIBCModule).
 		AddRoute(icahosttypes.SubModuleName, icaHostIBCModule).
 		AddRoute(ibctransfertypes.ModuleName, transferIBCModule).
-		AddRoute(intertxtypes.ModuleName, interTxIBCModule)
+		AddRoute(intertxtypes.ModuleName, icaControllerIBCModule)
```

Comment:
Sorry, can you explain me why this change?

Reply:
We need to route the packet to the top level of the middleware stack. See [docs](https://github.com/cosmos/ibc-go/blob/main/docs/ibc/middleware/integration.md#integrating-ibc-middleware-into-a-chain)

> An IBC application may be wrapped by many layers of middleware, only the top layer middleware should be hooked to the IBC router, with all underlying middlewares and application getting wrapped by it.